datadog/cmd/main/main.go

```diff
@@ -214,10 +215,15 @@ func (d *DatadogCostSource) getDDCostsForWindow(window opencost.Window, listPric
 	// this post processing stage de-duplicates those usages and costs
 	postProcess(&ccResp)
 
+	// query from the first of the window's month until the window end's day so that we can properly adjust for the
+	// cumulative nature of the response
+	startDate := time.Date(window.Start().Year(), window.Start().Month(), 1, 0, 0, 0, 0, window.Start().Location())
+	endDate := time.Date(window.End().Year(), window.End().Month(), window.End().Day(), 0, 0, 0, 0, window.End().Location())
```

Comment:
We have a lot of baked-in assumptions (in both Opencost and Kubecost) about times being UTC-aligned. I'd feel a little safer if all of these were constructed with UTC as the "location"

Reply:
I take that back, I can see how this could still cause problems.